test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  MetadataAgentConfiguration config;
+  static void ParseConfiguration(MetadataAgentConfiguration& config, const std::string configString) {
```

Comment:
Actually, now that you have an instance variable, you might as well remove the `config` parameter and let it refer to the `config` instance variable. Of course, this can then no longer be `static`.

Reply:
Done.